azurelinuxagent/pa/rdma/factory.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2016 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+import azurelinuxagent.common.logger as logger
+
+from azurelinuxagent.common.version import DISTRO_NAME
+from azurelinuxagent.common.rdma import RDMAHandler
+from .suse import SUSERDMAHandler
+
+
+def get_rdma_handler(distro_name=DISTRO_NAME, distro_version=DISTRO_VERSION):
```

Comment:
I missed the rdma -> common move, thanks for that. The factory in pa is really for the distros that need special provisioning logic. It looks like the driver install is done from the daemon, which would be invoked at run time, not provisioning time, so it seems more appropriate that this be done in osutil?